rpcs3/Emu/NP/np_requests.cpp

```diff
@@ -915,66 +1056,80 @@ namespace np
 
-		if (reply.is_error() || !resp->Verify(verifier))
+		if (reply.is_error())
 		{
-			rpcn_log.error("Error parsing response in reply_record_score");
+			rpcn_log.error("Error parsing response in handle_GetScoreResponse");
 			return false;
 		}
 
-		ensure(trans_ctx->data.size() == 7);
-		u32 arrayNum                              = trans_ctx->data[0];
-		vm::ptr<SceNpScoreRankData> rankArray     = vm::cast(trans_ctx->data[1]);
-		vm::ptr<SceNpScoreComment> commentArray   = vm::cast(trans_ctx->data[2]);
-		vm::ptr<void> infoArray                   = vm::cast(trans_ctx->data[3]);
-		bool info_array_is_SceNpScoreGameInfo     = trans_ctx->data[4];
-		vm::ptr<CellRtcTick> lastSortDate         = vm::cast(trans_ctx->data[5]);
-		vm::ptr<SceNpScoreRankNumber> totalRecord = vm::cast(trans_ctx->data[6]);
+		const auto* tdata = std::get_if<tdata_get_score_generic>(&trans_ctx->tdata);
+		ensure(tdata);
+		ensure(resp->rankArray() && resp->rankArray()->size() <= tdata->arrayNum);
 
-		ensure(resp->rankArray() && resp->rankArray()->size() <= arrayNum);
-
-		memset(rankArray.get_ptr(), 0, sizeof(SceNpScoreRankData) * arrayNum);
+		memset(tdata->rankArray.get_ptr(), 0, tdata->rankArraySize);
 		auto* fb_rankarray = resp->rankArray();
+		u32 target_index   = 0;
+
+		vm::ptr<SceNpScoreRankData> rankArray             = vm::static_ptr_cast<SceNpScoreRankData>(tdata->rankArray);
+		vm::ptr<SceNpScorePlayerRankData> rankPlayerArray = vm::static_ptr_cast<SceNpScorePlayerRankData>(tdata->rankArray);
+
 		for (flatbuffers::uoffset_t i = 0; i < fb_rankarray->size(); i++)
 		{
 			const auto* fb_rankdata = fb_rankarray->Get(i);
 			ensure(fb_rankdata->npId() && fb_rankdata->onlineName());
-			string_to_npid(fb_rankdata->npId()->string_view(), &rankArray[i].npId);
-			string_to_online_name(fb_rankdata->onlineName()->string_view(), &rankArray[i].onlineName);
-			rankArray[i].pcId            = fb_rankdata->pcId();
-			rankArray[i].serialRank      = fb_rankdata->rank();
-			rankArray[i].rank            = fb_rankdata->rank();
-			rankArray[i].highestRank     = fb_rankdata->rank();
-			rankArray[i].scoreValue      = fb_rankdata->score();
-			rankArray[i].hasGameData     = fb_rankdata->hasGameData();
-			rankArray[i].recordDate.tick = fb_rankdata->recordDate();
+
+			if (fb_rankdata->recordDate() == 0)
+				continue;
+
+			SceNpScoreRankData* cur_rank;
+			if (tdata->rankArraySize == (tdata->arrayNum * sizeof(SceNpScoreRankData)))
+			{
+				cur_rank = &rankArray[i];
+			}
+			else
+			{
+				rankPlayerArray[i].hasData = 1;
+				cur_rank                   = &rankPlayerArray[i].rankData;
+			}
+
+			string_to_npid(fb_rankdata->npId()->string_view(), &cur_rank->npId);
+			string_to_online_name(fb_rankdata->onlineName()->string_view(), &cur_rank->onlineName);
+
+			cur_rank->pcId            = fb_rankdata->pcId();
+			cur_rank->serialRank      = fb_rankdata->rank();
+			cur_rank->rank            = fb_rankdata->rank();
+			cur_rank->highestRank     = fb_rankdata->rank();
```

Comment:
Because not implemented on the server(serialRank is normal ranking, rank is ranking with people with the same score occupying the same rank and highestRank is redudant with rank).